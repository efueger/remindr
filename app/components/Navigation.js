import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
} from 'react-native';

import taskTypes from '../util/taskTypes';

export default function Navigation(props) {
  const types = [taskTypes.DEFERRED, taskTypes.CURRENT, taskTypes.DONE];

  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        {types.map((type) => {
          const buttonStyles = [styles.button];

          if (type.id === props.selectedType) {
            buttonStyles.push({
              backgroundColor: type.color,
            }, styles.selected);
          }

          // This is the most elegent way of binding the callback with the value.
          // eslint-disable-next-line react/jsx-no-bind
          return (
            <TouchableHighlight
              onPress={props.onPress.bind(null, { filterType: type.id })}
              key={type.id}
              accessibilityLabel={`navigation-${type.id.toLowerCase()}`}
            >
              <Text style={buttonStyles}>{type.name.toUpperCase()}</Text>
            </TouchableHighlight>
          );
        })}
      </View>
    </View>
  );
}

Navigation.propTypes = {
  onPress: React.PropTypes.func.isRequired,
  selectedType: React.PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    borderTopColor: '#d9dbdb',
    borderTopWidth: 1,
    backgroundColor: '#fff',
  },
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    padding: 12,
    fontSize: 11,
    fontWeight: '300',
  },
  selected: {
    color: '#fff',
  },
});
