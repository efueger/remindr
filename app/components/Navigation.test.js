import React from 'react';
import Navigation from './Navigation';
import renderer from 'react-test-renderer';

it('renders the navigation component', () => {
  const component = renderer.create(
    <Navigation selectedType={'CURRENT'} onPress={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
