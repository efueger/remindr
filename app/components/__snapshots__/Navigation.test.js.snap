exports[`test renders the navigation component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderTopColor": "#d9dbdb",
      "borderTopWidth": 1
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center"
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent"
          },
          undefined
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12
            }
          ]
        }>
        DEFERRED
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent"
          },
          undefined
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12
            },
            Object {
              "backgroundColor": "#03a9f4"
            },
            Object {
              "color": "#fff"
            }
          ]
        }>
        CURRENT
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent"
          },
          undefined
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "300",
              "padding": 12
            }
          ]
        }>
        DONE
      </Text>
    </View>
  </View>
</View>
`;
